### ISSUE TITLE
training stoped when performance continued decresing

### ISSUE THREAD

**Issue by zhangyike** [Microsoft/CNTK#1423]
I trained a language model using adaptive learning rate rule. However, when perplexity continuing increased in two epochs, training stopped. Why do not CNTK reduce learning rate and continue training? 
Here is my configure:
```
   SGD = [
        minibatchSize = 128
        learningRatesPerSample = 0.01
        momentumPerMB = 0
        maxEpochs = $epoch$
        numMBsToShowResult = 500
        
        gradientClippingWithTruncation = true
        clippingThresholdPerSample = 15.0

        loadBestModel = false
        dropoutRate = 0.0

        # settings for Auto Adjust Learning Rate
        AutoAdjust = [
            autoAdjustLR = "adjustAfterEpoch"
            reduceLearnRateIfImproveLessThan = 0.001
            learnRateDecreaseFactor = 0.5
            # continueReduce = false
            continueReduce = true
            increaseLearnRateIfImproveMoreThan = 1000000000
            learnRateIncreaseFactor = 1.382
        ]
    ]
```
And  training log:
```
Starting Epoch 1: learning rate per sample = 0.002000  effective momentum = 0.000000  momentum as time constant = 0.0 samples

Starting minibatch loop.
LMSequenceReader: Reading epoch data... 37416 sequences read.
WARNING: The same matrix with dim [4, 120] has been transferred between different devices for 20 times.
 Epoch[ 1 of 10]-Minibatch[   1- 500]: evalNodes = 512.497% * 55186; time = 23.4469s; samplesPerSecond = 2353.7
 Epoch[ 1 of 10]-Minibatch[ 501-1000]: evalNodes = 511.753% * 54994; time = 23.2376s; samplesPerSecond = 2366.6
 Epoch[ 1 of 10]-Minibatch[1001-1500]: evalNodes = 506.181% * 55197; time = 22.7825s; samplesPerSecond = 2422.8
 Epoch[ 1 of 10]-Minibatch[1501-2000]: evalNodes = 507.170% * 55139; time = 23.5353s; samplesPerSecond = 2342.8
 Epoch[ 1 of 10]-Minibatch[2001-2500]: evalNodes = 507.919% * 55159; time = 22.4553s; samplesPerSecond = 2456.4
 Epoch[ 1 of 10]-Minibatch[2501-3000]: evalNodes = 505.179% * 55269; time = 22.5203s; samplesPerSecond = 2454.2
 Epoch[ 1 of 10]-Minibatch[3001-3500]: evalNodes = 501.922% * 55241; time = 22.8503s; samplesPerSecond = 2417.5
 Epoch[ 1 of 10]-Minibatch[3501-4000]: evalNodes = 503.062% * 55240; time = 22.7170s; samplesPerSecond = 2431.7
 Epoch[ 1 of 10]-Minibatch[4001-4500]: evalNodes = 501.131% * 55282; time = 22.5696s; samplesPerSecond = 2449.4
 Epoch[ 1 of 10]-Minibatch[4501-5000]: evalNodes = 500.701% * 55314; time = 22.5810s; samplesPerSecond = 2449.6
 Epoch[ 1 of 10]-Minibatch[5001-5500]: evalNodes = 500.085% * 55313; time = 22.7159s; samplesPerSecond = 2435.0
 Epoch[ 1 of 10]-Minibatch[5501-6000]: evalNodes = 497.420% * 55059; time = 22.7041s; samplesPerSecond = 2425.1
LMSequenceReader: Reading epoch data... 0 sequences read.
Finished Epoch[ 1 of 10]: [Training] evalNodes = 504.535% * 683293; totalSamplesSeen = 683293; learningRatePerSample = 0.0020000001; epochTime=283.019s
LMSequenceReader: Reading epoch data... 503 sequences read.
LMSequenceReader: Reading epoch data... 0 sequences read.
Final Results: Minibatch[1-95]: evalNodes = 593.302% * 8837; perplexity = 377.29334661
Finished Epoch[ 1 of 10]: [Validate] evalNodes = 593.302% * 8837
SGD: Saving checkpoint model './Output_MT/Models_baseline/wsj.lstm.mdl.1'

Starting Epoch 2: learning rate per sample = 0.002000  effective momentum = 0.000000  momentum as time constant = 0.0 samples

Starting minibatch loop.
LMSequenceReader: Reading epoch data... 37416 sequences read.
 Epoch[ 2 of 10]-Minibatch[   1- 500, 9.69%]: evalNodes = 492.262% * 55066; time = 22.8234s; samplesPerSecond = 2412.7
 Epoch[ 2 of 10]-Minibatch[ 501-1000, 19.37%]: evalNodes = 492.962% * 55217; time = 22.6262s; samplesPerSecond = 2440.4
 Epoch[ 2 of 10]-Minibatch[1001-1500, 29.06%]: evalNodes = 491.972% * 55197; time = 22.5307s; samplesPerSecond = 2449.9
 Epoch[ 2 of 10]-Minibatch[1501-2000, 38.75%]: evalNodes = 492.652% * 55203; time = 22.7612s; samplesPerSecond = 2425.3
 Epoch[ 2 of 10]-Minibatch[2001-2500, 48.43%]: evalNodes = 490.469% * 55341; time = 22.3306s; samplesPerSecond = 2478.3
 Epoch[ 2 of 10]-Minibatch[2501-3000, 58.12%]: evalNodes = 487.789% * 55187; time = 22.6677s; samplesPerSecond = 2434.6
 Epoch[ 2 of 10]-Minibatch[3001-3500, 67.81%]: evalNodes = 487.627% * 55200; time = 22.8398s; samplesPerSecond = 2416.8
 Epoch[ 2 of 10]-Minibatch[3501-4000, 77.49%]: evalNodes = 486.698% * 55155; time = 22.9007s; samplesPerSecond = 2408.4
 Epoch[ 2 of 10]-Minibatch[4001-4500, 87.18%]: evalNodes = 487.651% * 55338; time = 22.6165s; samplesPerSecond = 2446.8
 Epoch[ 2 of 10]-Minibatch[4501-5000, 96.87%]: evalNodes = 486.962% * 55285; time = 24.6676s; samplesPerSecond = 2241.2
 Epoch[ 2 of 10]-Minibatch[5001-5500, 106.55%]: evalNodes = 487.647% * 55266; time = 22.6110s; samplesPerSecond = 2444.2
 Epoch[ 2 of 10]-Minibatch[5501-6000, 116.24%]: evalNodes = 483.868% * 55114; time = 24.3327s; samplesPerSecond = 2265.0
LMSequenceReader: Reading epoch data... 0 sequences read.
Finished Epoch[ 2 of 10]: [Training] evalNodes = 488.885% * 683293; totalSamplesSeen = 1366586; learningRatePerSample = 0.0020000001; epochTime=284.708s
LMSequenceReader: Reading epoch data... 503 sequences read.
LMSequenceReader: Reading epoch data... 0 sequences read.
Final Results: Minibatch[1-95]: evalNodes = 600.089% * 8837; perplexity = 403.78970072
Finished Epoch[ 2 of 10]: [Validate] evalNodes = 600.089% * 8837
learnRatePerSample reduced to 0.001
SGD: Saving checkpoint model './Output_MT/Models_baseline/wsj.lstm.mdl.2'

Starting Epoch 3: learning rate per sample = 0.001000  effective momentum = 0.000000  momentum as time constant = 0.0 samples

Starting minibatch loop.
LMSequenceReader: Reading epoch data... 37416 sequences read.
 Epoch[ 3 of 10]-Minibatch[   1- 500, 9.69%]: evalNodes = 478.551% * 55075; time = 22.6120s; samplesPerSecond = 2435.7
 Epoch[ 3 of 10]-Minibatch[ 501-1000, 19.37%]: evalNodes = 476.399% * 55299; time = 22.5634s; samplesPerSecond = 2450.8
 Epoch[ 3 of 10]-Minibatch[1001-1500, 29.06%]: evalNodes = 473.702% * 55118; time = 22.6024s; samplesPerSecond = 2438.6
 Epoch[ 3 of 10]-Minibatch[1501-2000, 38.75%]: evalNodes = 474.366% * 55287; time = 22.6441s; samplesPerSecond = 2441.6
 Epoch[ 3 of 10]-Minibatch[2001-2500, 48.43%]: evalNodes = 472.311% * 55212; time = 22.6861s; samplesPerSecond = 2433.7
 Epoch[ 3 of 10]-Minibatch[2501-3000, 58.12%]: evalNodes = 471.328% * 55313; time = 23.2011s; samplesPerSecond = 2384.1
 Epoch[ 3 of 10]-Minibatch[3001-3500, 67.81%]: evalNodes = 474.780% * 55250; time = 23.3781s; samplesPerSecond = 2363.3
 Epoch[ 3 of 10]-Minibatch[3501-4000, 77.49%]: evalNodes = 471.017% * 55144; time = 22.6361s; samplesPerSecond = 2436.1
 Epoch[ 3 of 10]-Minibatch[4001-4500, 87.18%]: evalNodes = 473.541% * 55286; time = 22.9743s; samplesPerSecond = 2406.4
 Epoch[ 3 of 10]-Minibatch[4501-5000, 96.87%]: evalNodes = 473.800% * 55242; time = 24.0049s; samplesPerSecond = 2301.3
 Epoch[ 3 of 10]-Minibatch[5001-5500, 106.55%]: evalNodes = 472.232% * 55125; time = 23.6426s; samplesPerSecond = 2331.6
 Epoch[ 3 of 10]-Minibatch[5501-6000, 116.24%]: evalNodes = 472.304% * 55181; time = 22.5042s; samplesPerSecond = 2452.0
LMSequenceReader: Reading epoch data... 0 sequences read.
Finished Epoch[ 3 of 10]: [Training] evalNodes = 473.540% * 683293; totalSamplesSeen = 2049879; learningRatePerSample = 0.001; epochTime=284.422s
LMSequenceReader: Reading epoch data... 503 sequences read.
LMSequenceReader: Reading epoch data... 0 sequences read.
Final Results: Minibatch[1-95]: evalNodes = 603.146% * 8837; perplexity = 416.32336650
Finished Epoch[ 3 of 10]: [Validate] evalNodes = 603.146% * 8837
Finished training and saved final model
```

**Comment by zhangyike** [comment]
@vmazalov @dongyu888 Could you please give me any advices?

**Comment by sayanpa** [comment]
CNTK allows for controlling your learning rate as well as custom exit logic for early stopping. We have several options in the CNTK/Manuals directory. You may want to start with removing Auto Adjust.

Given this issue has been open for so long I am closing it. But feel free to reopen and new one if you need help.
